src/App.jsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import { Clock } from './components/Clock/Clock';
+import './App.scss';
+
+export class App extends React.Component {
+  state = {
+    isClockVisible: true,
+    clockName: 'new name',
+    oldClockName: 'old name',
+  }
+
+  hideClock = () => {
+    this.setState({
+      isClockVisible: false,
+    });
+  }
+
+  showClock = () => {
+    this.setState({
+      isClockVisible: true,
+    });
+  }
+
+  setRandomName = () => {
+    this.setState(prevState => (
+      {
+        oldClockName: prevState.clockName,
+        clockName: Math.ceil(Math.random() * 10),
+      }
+    ));
+    // eslint-disable-next-line no-console
+    console.log(
+      `The Clock was renamed from
+      ${this.state.oldClockName} to ${this.state.clockName}`,
+    );
+  }
+
+  render() {
+    return (
+      <div className="App">
+        <h1>React clock</h1>
+
+        <p>
+          Current time:
+          {' '}
+
+          <Clock
+            isClockVisible={this.state.isClockVisible}
+          />
```

Comment:
let's use conditional rendering here instead of passing isClockVisible to the Clock component